src/block/PinkPetals.php

```diff
@@ -98,15 +99,9 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 
 	private function grow(?Player $player) : bool{
 		if($this->count < self::MAX_COUNT){
-			$ev = new BlockGrowEvent($this, (clone $this)->setCount($this->count + 1), $player);
-			$ev->call();
-			if($ev->isCancelled()){
-				return false;
-			}
-			$this->position->getWorld()->setBlock($this->position, $ev->getNewState());
-		}else{
-			$this->position->getWorld()->dropItem($this->position->add(0, 0.5, 0), $this->asItem());
+			return BlockEventHelper::grow($this, (clone $this)->setCount($this->count + 1), $player);
 		}
+		$this->position->getWorld()->dropItem($this->position->add(0, 0.5, 0), $this->asItem());
```

Comment:
I'm not sure I understand why this is in the grow() function.

Reply:
On closer look the grow function seems completely redundant, since there's no random-tick growth.

However, if there was random-tick growth, I wouldn't expect this to be dropping items, otherwise we'd get random dropped items scattered across the world.